plugins/dominant-color-images/hooks.php

```diff
@@ -16,13 +16,17 @@
  *
  * @since 1.0.0
  *
- * @param array $metadata      The attachment metadata.
- * @param int   $attachment_id The attachment ID.
- * @return array $metadata The attachment metadata.
+ * @param array|mixed $metadata      The attachment metadata.
+ * @param int         $attachment_id The attachment ID.
+ * @return array{ has_transparency?: bool, dominant_color?: string } The amended attachment metadata.
  */
-function dominant_color_metadata( $metadata, $attachment_id ) {
+function dominant_color_metadata( $metadata, int $attachment_id ): array {
+	if ( ! is_array( $metadata ) ) {
+		$metadata = array();
+	}
```

Comment:
Checks like this and the use of `mixed` as a possible param type are needed because plugins can add filters that stick any type in for `$metadata`.